p2p/pubsub/wrapper.go

```diff
@@ -34,6 +36,19 @@ func (ps *PubSub) Register(topic string, handler GossipHandler) {
 		rst := handler(log.WithNewRequestID(ctx), pid, msg.Data)
 		metrics.ProcessedMessagesDuration.WithLabelValues(topic, castResult(rst)).
 			Observe(float64(time.Since(start)))
+		if rst == ValidationReject {
+			// We want to disconnect the peer and also penalize it which could result in it being blacklisted.
+
+			// Ok we close the peer here but the deadpeerhandler may try a redial if the state of the conn is connected
+			err := ps.host.Network().ClosePeer(pid)
+			if err != nil {
+				ps.logger.With().Error("failed to close peer",
```

Comment:
this should be Debug. for information maybe it makes sense to add a counter how many peers are disconnected

but also this approach is not sufficient, as mentioned we need to drop peer also when it gives bad message over sync. thats why i suggested to decorate handlers

Reply:
@dshulyak I've made the changes we discussed, please take a look.